lib/db/dao/quiz_progress_dao.dart

```diff
@@ -4,26 +4,8 @@ import 'package:maui/db/entity/quiz_progress.dart';
 import 'package:sqflite/sqflite.dart';
 
 class QuizProgressDao {
-  Future<QuizProgress> getQuizProgressByQuizProgressId(String id, {Database db}) async {
-    db = db ?? await new AppDatabase().getDb();
-    List<Map> maps = await db.query(QuizProgress.table,
-        columns: [
-          QuizProgress.idCol,
-          QuizProgress.userIdCol,
-          QuizProgress.topicIdCol,
-          QuizProgress.quizIdCol,
-          QuizProgress.maxScoreCol,
-          QuizProgress.outOfTotalCol
-        ],
-        where: "${QuizProgress.idCol} = ? ",
-        whereArgs: [id]);
-    if (maps.length > 0) {
-      return QuizProgress.fromMap(maps.first);
-    }
-    return null;
-  }
 
-  Future<QuizProgress> getQuizProgressByTopicId(String topicId, {Database db}) async {
+  Future<double> getQuizProgressByTopicId(String topicId, {Database db}) async {
```

Comment:
Change name to something more relevant like getScoreSummaryByTopicId. getQuizProgressByTopicId assumes you will be returning back a QuizProgress object